plugins/modules/gitlab_project_members.py

```diff
@@ -316,8 +404,12 @@ def main():
         gitlab_users_access = module.params['gitlab_users_access']
         for user_level in gitlab_users_access:
             user_level['access_level'] = access_level_int[user_level['access_level']]
-
-    if len(gitlab_users_access) == 1 and not purge_users:
+    elif module.params['gitlab_group'] is not None:
+        gitlab_groups = module.params['gitlab_group']
+        groups = project.get_groups_in_a_project(gitlab_project_id)
+    if groups:
```

Comment:
This condition looks wrong to me. If `module.params['gitlab_group'] is not None` but `groups` happens to be empty, the below cases are handled, and eventually the module exits with "Nothing to do, please give at least one user or group or set purge_users true.". That doesn't sound right to me.

Reply:
This problem is still there. Just because the module argument is not `None` doesn't mean the list is not empty.